### PR TITLE
docs(encoding): add examples for json docs

### LINKED ISSUE #93: json

### PR DESCRIPTION
**Pull request by ramfox**
closes #93

### CODE REVIEW

File: encoding/json/doc.go [hunk 1]
```diff
@@ -7,43 +7,68 @@ go.starlark.net/starlarkjson
...
+      examples:
+        encode object
+          encode a simple object as a JSON string
+          code:
+            load("encoding/json.star", "json")
+            x = json.encode({"foo": ["bar", "baz"]})
+            print(x)
+            # Output: {"foo":["bar","baz"]}
+    decode(src) obj
+      Return the Starlark representation of a string instance containing a JSON document. Decoding fails if src is not a valid JSON string.
```
Comment (ramfox) [inline encoding/json/doc.go:28]: took phrasing from python docs for `decode`

File: encoding/json/doc.go [hunk 2]
```diff
@@ -7,43 +7,68 @@ go.starlark.net/starlarkjson
...
-        and returns a string containing the indented form.
-        It accepts one required positional parameter, the JSON string,
-        and two optional keyword-only string parameters, prefix and indent,
-        that specify a prefix of each new line, and the unit of indentation.
+outline: json
+  json provides functions for working with json data
+  path: encoding/json
+  functions:
+    encode(obj) string
+      Return a JSON string representation of a Starlark data structure
```
Comment (ramfox) [inline encoding/json/doc.go:15]: took phrasing from python docs for `encode`

File: encoding/json/doc.go [hunk 3]
```diff
@@ -7,43 +7,68 @@ go.starlark.net/starlarkjson
...
+					code:
+						load("encoding/json.star", "json")
+            x = json.indent('{"foo": ["bar", "baz"]}', prefix='....', indent="____")
+						# print(x)
+						# {
+						# ....____"foo": [
+						# ....________"bar",
+						# ....________"baz"
+						# ....____]
+						# ....}
```
Comment (chriswhong) [inline encoding/json/doc.go:71]: image

Would indent really prefix the closing bracket but not the opening bracket?

**Review by chriswhong — APPROVED** [review]
One question about `indent`, then LGTM

File: encoding/json/doc.go (hunk 3 above)
Comment (ramfox) [reply encoding/json/doc.go:71]: that's how it is currently happening (This method uses golang's `json.MarshalIndent` under the hood.)! whether or not that's the desired behavior is another question.

**Pull request merged**